### ISSUE TITLE
`Statement is unreachable` after unconditional raise

### ISSUE THREAD

**Issue by opk12** [python/mypy#13037]
Sometimes I raise `NotImplementedError` or `AssertionError` at the beginning of a function, to ensure that it never runs. I use this for the half-finished functions, that are not ready for production, but cannot be deleted and even end up in the end-user software package, because they
- centralize documentation about the issues that prevent a full implementation
- or wait to be refactored into something functional
- or wait requirements reanalysis in another development iteration

We can add `type: ignore`, but the intention of raising unconditionally is clear. Should mypy not warn that `Statement is unreachable`, or is there an idiomatic approach that does not trigger mypy?

```
$  cat a.py
def f() -> None:
    raise NotImplementedError("reason")  # or AssertionError
    print("")

$  mypy --warn-unreachable a.py
a.py:3: error: Statement is unreachable
Found 1 error in 1 file (checked 1 source file)
```

**Comment by AlexWaygood** [comment]
How about:

```python
from typing import TYPE_CHECKING

def f() -> None:
    if not TYPE_CHECKING:
        raise NotImplementedError("reason")  # or AssertionError
    print("")
```

?

**Comment by hauntsaninja** [comment]
I think either explicitly ignore with `# type: ignore[unreachable]`, or don't use `--warn-unreachable`
